### PR TITLE
Upgrade to OpenTelemetry 1.0 RC3

### PR DESCRIPTION
**Pull request by maciej-szlosarczyk**
This version brings a couple of changes, most of which are
improvements:

- You no longer need to register an application tracer, this is done automatically on library level.
- :otel_propagator.text_map.extract/1 becomes :otel_propagator_text_map.extract/1
- Default tracer version becomes 1.0.0-rc.3

This should be released as a tag with version.

Oh yeah and it fixes the error with missing top-level traces when using sampler 🎉 

Signed-off-by: Maciej Szlosarczyk <maciej@mindvalley.com>

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **maciej-szlosarczyk** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: mix.exs [hunk 1]
```diff
@@ -37,11 +37,12 @@ defmodule MvOpentelemetry.MixProject do
   defp deps do
     [
       # OpenTelemetry
-      {:opentelemetry, "~> 0.6.0", override: true},
```
Comment (maciej-szlosarczyk) [inline mix.exs:40]: We actually only need this in dev/test

File: lib/mv_opentelemetry.ex [hunk 2]
```diff
@@ -32,16 +31,6 @@ defmodule MvOpentelemetry do
   ```
   """
 
-  @doc """
-  Registers an application tracer for your OTP application. Should be called only once
-  per application startup.
-  """
-  @spec register_application(atom()) :: :ok
```
Comment (maciej-szlosarczyk) [inline lib/mv_opentelemetry.ex:39]: I'm thinking maybe to retain it for backwards compatibility, but only stories and insights use it at the moment so who cares, I can remove this code.

File: README.md [hunk 3]
```diff
@@ -91,7 +90,14 @@ config :opentelemetry,
   processors: [
     otel_batch_processor: %{
       exporter:
-        {OpenTelemetry.Honeycomb.Exporter, write_key: "REPLACE_ME", dataset: "REPLACE_ME"}
+        {:opentelemetry_exporter,
+         %{
+           endpoints: [{:https, 'api.honeycomb.io', 443, '/v1/traces'}],
```
Comment (maciej-szlosarczyk) [inline README.md:95]: This is actually super great, we can remove the rather problematic/old Honeycomb library and use default exporter, honeycomb will accept it.

**Review by Vollmond — APPROVED** [review]

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: a77879623d0600a146fab30f30bf88236f64538f</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/11#pullrequestreview-796172193" title="Approved">Vollmond</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [Vollmond]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**